velox/functions/prestosql/tests/TimestampWithTimeZoneCastTest.cpp

```diff
@@ -64,6 +65,74 @@ TEST_F(TimestampWithTimeZoneCastTest, fromTimestamp) {
   testCast(tsVector, expected);
 }
 
+TEST_F(TimestampWithTimeZoneCastTest, fromVarchar) {
+  const auto stringVector = makeNullableFlatVector<StringView>(
+      {std::nullopt,
+       "2012-10-31 01:00:47 America/Denver",
+       "2012-10-31 01:00:47 -06:00",
+       "1994-05-06 15:49 Europe/Vienna",
+       "1979-02-24 08:33:31 Pacific/Chatham",
+       "1979-02-24 08:33:31 +13:45"});
+
+  // Varchar representations above hold local time (relative to specified time
+  // zone). For instance, the first string represents a wall-clock displaying
+  // '2012-10-31 01:00:47' in Denver. Below, we use UTC representations of the
+  // above local wall-clocks, to match the UTC timepoints held in the
+  // TimestampWithTimezone type.
+  auto denverUTC = util::fromTimestampString("2012-10-31 07:00:47").toMillis();
+  auto viennaUTC = util::fromTimestampString("1994-05-06 13:49:00").toMillis();
+  auto chathamUTC = util::fromTimestampString("1979-02-23 18:48:31").toMillis();
+
+  auto timestamps = std::vector<int64_t>{
+      0, denverUTC, denverUTC, viennaUTC, chathamUTC, chathamUTC};
+
+  auto timezones = std::vector<TimeZoneKey>{
+      {0,
+       (int16_t)util::getTimeZoneID("America/Denver"),
+       (int16_t)util::getTimeZoneID("-06:00"),
+       (int16_t)util::getTimeZoneID("Europe/Vienna"),
+       (int16_t)util::getTimeZoneID("Pacific/Chatham"),
+       (int16_t)util::getTimeZoneID("+13:45")}};
+
+  const auto expected = makeTimestampWithTimeZoneVector(
```

Comment:
same here. You can use evaluateOnce<int64_t>(), then unpack the result.

Reply:
Swapped.